modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -667,7 +677,22 @@
                     (and (= (temporal-type hsql-form) :timestamp)
                          (not (contains? (temporal-type->supported-units :timestamp) unit)))
                     (hx/cast :datetime))]
-    (AddIntervalForm. hsql-form amount unit)))
+    (add-interval-form hsql-form amount unit)))
+
+(defn x []
+  (mt/with-driver :bigquery-cloud-sdk
+    (mt/dataset office-checkins
+      (mt/with-temp-vals-in-db 'Field (mt/id :checkins :timestamp) {:database_type "TIMESTAMP"}
```

Comment:
For some annoying reason `office-checkins.checkins.timestamp` is actually a `DATETIME` for BigQuery so I have to fake making it a `TIMESTAMP` here.